python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -610,3 +610,95 @@ def where(array):
         return np.where(array)
     else:
         raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+def take(array, indices, axis: int):
+    """
+    See torch take_along_dim or numpy take as reference.
+
+    Because numpy and torch have different APIs we went for the more limited one
+    in torch not supporting ``mode`` argument. The argument ``out`` is not supported
+    by TorchScript, or at least it is nontrivial to add
+
+    :param array: the array the elements are returned from
+    :param indices: the indices to take long an axes
+    :param axis: axis of array to take
+
+    :return: the elments at the indices in the array along dimension specified by the
+    axis
+    """
+    if isinstance(array, TorchTensor):
+        return torch.index_select(array, dim=axis, index=indices)
+    elif isinstance(array, np.ndarray):
+        return np.take(array, indices, axis=axis)
+    else:
+        raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+# PR COMMENT I implemented argsort because I thought I need it, but it turned out not
+#            but I would nevertheless keep it, might be useful for something else
+def argsort(array, axis=-1, descending: bool = False, stable: bool = False):
+    """
+    See torch argsort as reference.
+
+    Because numpy and torch have different APIs we went for the more limited one
+    in torch. Numpy supports more options for sorting that are not here available.
+
+    :param array: the array that is argsorted
+    :param axis: axis of array to argsort
+    :param stable: if true, the order of duplicate elements stays the same an the array
+    :param descending: if false, the order is ascending
+
+    :return: indices of the sorted array wrt. the unsorted one
+    """
+    if isinstance(array, TorchTensor):
+        return torch.argsort(array, dim=axis, descending=descending, stable=stable)
+    elif isinstance(array, np.ndarray):
+        if stable:
+            idx_ascending = np.argsort(array, axis=axis, kind="stable")
+        else:
+            idx_ascending = np.argsort(array, axis=axis, kind="quicksort")
+
+        if descending:
+            return np.flip(idx_ascending, axis=axis)
+        else:
+            return idx_ascending
+    else:
+        raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+def argsort_metadata_values(metadata_values, reverse: bool = False):
```

Comment:
If my understanding is correct (see function summary below), then we can also do something like this
```suggestion
def argsort_labels_values(labels: Labels, reverse: bool = False):
```